matrices.tex

```diff
@@ -25,6 +25,8 @@
 {x - sqrt(y*(x + y)): 1, x + sqrt(y*(x + y)): 1}
 \end{verbatim}
 
+%% TODO a Python list is not a dense representation...
+%% Is this supposed to say flat? - scopatz
```

Comment:
I added a footnote to the polys section explaining what they mean (polys because the polys section comes before the matrix section). Hopefully it is sufficient to also clarify the term for the subsequent matrices section, even though the strict definition is slightly different (a matrix has entries and shape; a polynomial has coefficients and degree).